src/runtime/agas/big_boot_barrier.cpp

```diff
@@ -389,11 +458,19 @@ void register_worker(registration_header const& header)
     boost::uint32_t first_core = rt.assign_cores(header.hostname,
         header.cores_needed);
 
+    std::map<std::string, boost::uint32_t> desc_map;
+    std::for_each(header.typeids.cbegin(), header.typeids.cend(),
+        [&](const std::string& str){
+            desc_map.insert(std::make_pair(
+                str, unique_typeid_registry::get_type_descriptor(str))
+            );
+        });
+
     big_boot_barrier & bbb = get_big_boot_barrier();
 
     notification_header hdr (prefix, bbb.here(), locality_addr, primary_addr
       , component_addr, symbol_addr, rt.get_config().get_num_localities()
-      , first_core, bbb.get_endpoints());
+      , first_core, bbb.get_endpoints(), desc_map);
 
```

Comment:
We might not have to send back the typeids (strings) as long as we are able to keep the list of strings around at the requesting end and we make sure that the sequence of ids sent back is correct.